tests/cli/functional/conftest.py

```diff
@@ -16,18 +16,99 @@
+    MOCK_KEYSTORE_PATH,
+    MOCK_PROVIDER_URI,
+    NUMBER_OF_MOCK_KEYSTORE_ACCOUNTS
+)
 from tests.fixtures import _make_testerchain, make_token_economics
-from tests.mock.agents import FAKE_RECEIPT, MockContractAgency
+from tests.mock.agents import MockContractAgency, MockContractAgent
 from tests.mock.interfaces import MockBlockchain, make_mock_registry_source_manager
+from tests.utils.config import make_alice_test_configuration, make_bob_test_configuration, \
+    make_ursula_test_configuration
+from tests.utils.ursula import MOCK_URSULA_STARTING_PORT
+
+
+@pytest.fixture(scope='function', autouse=True)
+def mock_contract_agency(monkeypatch, module_mocker, token_economics):
+    monkeypatch.setattr(ContractAgency, 'get_agent', MockContractAgency.get_agent)
+    module_mocker.patch.object(EconomicsFactory, 'get_economics', return_value=token_economics)
+    mock_agency = MockContractAgency()
+    yield mock_agency
+    mock_agency.reset()
+
+
+@pytest.fixture(scope='function', autouse=True)
+def mock_token_agent(mock_testerchain, token_economics, mock_contract_agency):
+    return mock_contract_agency.get_agent(NucypherTokenAgent)
+
+
+@pytest.fixture(scope='function', autouse=True)
+def mock_staking_agent(mock_testerchain, token_economics, mock_contract_agency):
+    mock_agent = mock_contract_agency.get_agent(StakingEscrowAgent)
+    return mock_agent
+
+
+@pytest.fixture(scope='function', autouse=True)
+def mock_adjudicator_agent(mock_testerchain, token_economics, mock_contract_agency):
+    mock_agent = mock_contract_agency.get_agent(AdjudicatorAgent)
+    return mock_agent
+
+
+@pytest.fixture(scope='function', autouse=True)
+def mock_policy_manager_agent(mock_testerchain, token_economics, mock_contract_agency):
+    mock_agent = mock_contract_agency.get_agent(PolicyManagerAgent)
+    return mock_agent
+
+
+@pytest.fixture(scope='function', autouse=True)
+def mock_multisig_agent(mock_testerchain, token_economics, mock_contract_agency):
+    mock_agent = mock_contract_agency.get_agent(MultiSigAgent)
+    return mock_agent
+
+
+@pytest.fixture(scope='function', autouse=True)
+def mock_worklock_agent(mock_testerchain, token_economics, mock_contract_agency):
+    economics = token_economics
+
+    mock_agent = mock_contract_agency.get_agent(WorkLockAgent)
+
+    # Customize the mock agent
+    mock_agent.boosting_refund = economics.worklock_boosting_refund_rate
+    mock_agent.slowing_refund = 1  # TODO: another way to get this value?
+    mock_agent.start_bidding_date = economics.bidding_start_date
+    mock_agent.end_bidding_date = economics.bidding_end_date
+    mock_agent.end_cancellation_date = economics.cancellation_end_date
+    mock_agent.minimum_allowed_bid = economics.worklock_min_allowed_bid
+    mock_agent.lot_value = economics.worklock_supply
+
+    yield mock_agent
+    mock_agent.reset()
+
+
+@pytest.fixture(scope='function')
+def mock_click_prompt(mocker):
+    return mocker.patch.object(click, 'prompt')
```

Comment:
Just curious, why mock separate `click` methods instead of just mocking `stdin`? Is there something that can't be done this way?

Reply:
I just felt is was not necessary to test any deeper than `click.confirm` in a effort trying to precisely target layers StdIn -> Click.  This way we dont test `click` itself handling input.

Reply:
Yes, pretty much. I think it will be simpler and close to mimicking an actual user input, unless I'm missing some prompts where it wouldn't work.

Reply:
Yeah - this sounds like an excellent Idea to me, further isolating our application code from click at this test layer.  Let's address this in a follow up PR.  Hopefully this can also fix some of the flakiness of some tests that catch too much stdout currently.